src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -2203,5 +2203,86 @@ private int getExactClasses(CORINFO_CLASS_STRUCT_* baseType, int maxExactClasses
             Debug.Assert(index <= maxExactClasses);
             return index;
         }
+
+        private bool getReadonlyStaticFieldValue(CORINFO_FIELD_STRUCT_* fieldHandle, byte* buffer, int bufferSize)
+        {
+            Debug.Assert(fieldHandle != null);
+            Debug.Assert(buffer != null);
+            Debug.Assert(bufferSize > 0);
+
+            FieldDesc field = HandleToObject(fieldHandle);
+            if (field.IsStatic && !field.IsThreadStatic && field.IsInitOnly && field.OwningType is MetadataType owningType)
```

Comment:
Should this get the same check for the bufferSize as the JIT side?

Reply:
Do you mean `if (varTypeIsIntegral(lclTyp) || varTypeIsFloating(lclTyp) || (lclTyp == TYP_REF))` ? Do we need them here since the current impl is pretty robust and is able to return value for any data it supports `GetRawData` for

Reply:
Or you mean `field.GetFieldSize() == bufferSize` ? I don't see such API in NativeAOT part

Reply:
Added ` ((field.FieldType.IsPrimitive) || field.FieldType.IsObject))`

Reply:
Per my other comment, I think it would be reasonable to depend on the JIT for the basic eligibility checks and not duplicating the checks.